plugins/meta/flannel/flannel.go

```diff
@@ -241,6 +248,119 @@ func cmdAdd(args *skel.CmdArgs) error {
 	return delegateAdd(args.ContainerID, n.DataDir, n.Delegate)
 }
 
+func cmdAddWindows(containerID string, n *NetConf, fenv *subnetEnv) error {
+	if n.WindowsDelegate == nil {
+		n.WindowsDelegate = make(map[string]interface{})
+	} else {
+		if hasKey(n.WindowsDelegate, "type") && !isString(n.WindowsDelegate["type"]) {
+			return fmt.Errorf("'windowsDelegate' dictionary, if present, must have (string) 'type' field")
+		}
+		if hasKey(n.WindowsDelegate, "name") {
+			return fmt.Errorf("'windowsDelegate' dictionary must not have 'name' field, it'll be set by flannel")
+		}
+		if hasKey(n.WindowsDelegate, "ipam") {
+			return fmt.Errorf("'windowsDelegate' dictionary must not have 'ipam' field, it'll be set by flannel")
+		}
+	}
+
+	n.WindowsDelegate["name"] = n.Name
+
+	if !hasKey(n.WindowsDelegate, "type") {
+		n.WindowsDelegate["type"] = "wincni.exe"
+	}
+
+	updateOutboundNat(&n.WindowsDelegate, fenv)
```

Comment:
In Go, a `map` is already a reference type, there's rarely a reason to pass around a pointer to a map.

Reply:
Fixed